xarray/tests/test_units.py

```diff
@@ -7,12 +7,13 @@
 import xarray as xr
 from xarray.core import formatting
 from xarray.core.npcompat import IS_NEP18_ACTIVE
+from .test_variable import VariableSubclassobjects
 
 pint = pytest.importorskip("pint")
 DimensionalityError = pint.errors.DimensionalityError
 
 
-unit_registry = pint.UnitRegistry()
+unit_registry = pint.UnitRegistry(force_ndarray=True)
```

Comment:
Will do. Here's a more detailed explanation:

Because `Quantity` implements `__array_function__`, `xarray` treats `pint.Quantity` instances as arrays. This is mostly correct, but `pint` also allows wrapping python scalars which don't implement `ndim`, `shape` and `dtype`. Setting `force_ndarray` means `Quantity` instances always wrap `numpy.array`s and we don't get `AttributeError`s.

See hgrecco/pint#950.

Reply:
:+1: we should stick this in the docs somewhere.

Reply:
that's true. I will add that to the documentation todo item in #3594